mods/default/functions.lua

```diff
@@ -117,32 +118,62 @@ minetest.register_abm({
 	end,
 })
 
+
 --
 -- Papyrus and cactus growing
 --
 
+function default.grow_cactus(pos, node)
+	if node.param2 ~= 0 then
+		return
+	end
+	pos.y = pos.y-1
+	if minetest.get_item_group(minetest.get_node(pos).name, "sand") ~= 0 then
```

Comment:
`node.name`

Also, invert the condition (`if ... == 0 then return end`) to shorten lines.